src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -414,9 +440,14 @@ def get_result_set(
                         to_pandas,
                         to_iter and (i == len(plan.queries) - 1),
                         is_ddl_on_temp_object=query.is_ddl_on_temp_object,
+                        block=block,
+                        data_type=data_type,
                         **kwargs,
                     )
-                    placeholders[query.query_id_place_holder] = result["sfqid"]
+                    if block:
```

Comment:
maybe 
```
placeholders[query.query_id_place_holder] = result["sfqid"] if block else result.query_id
```

Reply:
I tried this one. But if return value is a asyncjob instance . will raise error that 'asyncjob class is not subscribable'

Reply:
no this comments means you can simplify your code, no functional change. So if you get an error, it means your original code is not correct.

Reply:
got it. turns out I got a wrong if last time.